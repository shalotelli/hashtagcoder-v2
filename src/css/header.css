.navigation li {
  @apply mr-5;
}

.navigation li a.hoverable:hover {
  @apply border-b border-gray-400 border-dashed;
}

.react-toggle-track {
  @apply bg-gray-400;
}

.react-toggle:hover:not(.react-toggle--disabled) .react-toggle-track {
  @apply bg-gray-800;
}

.react-toggle--checked .react-toggle-track {
  @apply bg-blue-500;
}

.react-toggle--checked:hover:not(.react-toggle--disabled) .react-toggle-track {
  @apply bg-blue-700;
}

.react-toggle-thumb {
  @apply border-gray-500;
}

.react-toggle--checked .react-toggle-thumb {
  @apply border-blue-500;
}