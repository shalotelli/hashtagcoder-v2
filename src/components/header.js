import React, {useState, useContext} from 'react'
import PropTypes from 'prop-types'
import {Link, useStaticQuery, graphql} from 'gatsby'
import Toggle from 'react-toggle'
import {ThemeContext} from '../context/theme-context'

import 'react-toggle/style.css'
import '../css/header.css'
import logo from '../images/hashtagcoder-icon.png'

import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import {
  faTwitter, 
  faInstagram, 
  faLinkedin, 
  faGithub,
  faYoutube
} from '@fortawesome/free-brands-svg-icons'
import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons'

const socialIconsMap = {
  twitter: faTwitter,
  instagram: faInstagram,
  linkedin: faLinkedin,
  github: faGithub,
  youtube: faYoutube
}

export const PureHeader = ({data, siteTitle}) => {
  const [showNav, setShowNav] = useState(false)
  const [navAnimation, setNavAnimation] = useState('fadeInRight')
  const {theme, setTheme} = useContext(ThemeContext)

  const toggleShowNav = () => {
    let animation = 'fadeIn'
    let timeout = 100

    if (showNav) {
      animation = 'fadeOut'
      timeout = 500
    } else {
      animation = 'fadeIn'
      timeout = 100
    }
    
    setNavAnimation(animation)
    setTimeout(() => setShowNav(!showNav), timeout)
  }

  const handleThemeToggle = () => {
    setTheme(theme === 'light' ? 'dark' : 'light')
  }

  return (
    <div className="absolute top-0 z-10 w-full">
      <header className="p-2 sm:p-8 sm:flex justify-between items-center">
        <div className="mb-3 sm:mb-0 flex justify-between items-center">
          <div className="text-2xl sm:text-3xl m-0">
            <Link to="/" className="flex items-center">
              <img src={logo} alt="logo" className="w-10 mr-1" />
              {siteTitle}
            </Link>
          </div>

          <div className="sm:hidden">
            <button type="button" 
              className="hover:text-gray-500 focus:text-gray-500 focus:outline-none"
              onClick={() => toggleShowNav()}>
              <FontAwesomeIcon icon={showNav ? faTimes : faBars} />
            </button>
          </div>
        </div>

        <ul className={`navigation bg-gray-200 rounded sm:bg-white p-2 sm:p-0 sm:flex animated ${navAnimation} ${showNav ? 'relative' : 'hidden'}`}>
          {data.appConfig.navigation.filter(x => x.active).map(link => (
            <li className="mb-2 sm:mb-0" key={link.page}>
              <Link to={`/${link.href}`} 
                className="hoverable" 
                activeClassName="border-b border-dashed">
                {link.value}
              </Link>
            </li>
          ))}

          <div className="flex">
            {data.appConfig.socialLinks.map((link, i) => (
              <li key={i}>
                <a href={link.href} 
                  target="_blank"
                  rel="noopener noreferrer">
                  <FontAwesomeIcon icon={socialIconsMap[link.icon]} />
                </a>
              </li>
            ))}
          </div>

          <Toggle
            id="theme-toggle"
            checked={theme === 'light'}
            onChange={handleThemeToggle}
      />
        </ul>
      </header>
    </div>
  )
}

const Header = ({siteTitle}) => {
  const data = useStaticQuery(graphql`
    {
      appConfig {
        socialLinks {
          href
          icon
        }
        navigation {
          value
          page
          href
          active
        }
      }
    }
  `)

  return <PureHeader data={data} siteTitle={siteTitle} />
}

Header.propTypes = {
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
