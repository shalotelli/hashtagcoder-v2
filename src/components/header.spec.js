import React from 'react'
import renderer from 'react-test-renderer'

import {PureHeader as Header} from './header'

describe('Header', () => {
  const data = {
    appConfig: {
      socialLinks: [
        {
          href: '',
          icon: 'twitter'
        }
      ],
      navigation: [
        {
          value: '',
          page: '',
          href: '',
          active: true
        }
      ]
    }
  }

  const component = renderer
      .create(<Header siteTitle="Hashtag Coder" 
        data={data} />)

  it('should render', () => {
      expect(component.toJSON()).toMatchSnapshot()
  })
  
})