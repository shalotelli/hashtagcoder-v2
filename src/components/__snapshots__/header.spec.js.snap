// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div
  className="absolute top-0 z-10 w-full"
>
  <header
    className="p-2 sm:p-8 sm:flex justify-between items-center"
  >
    <div
      className="mb-3 sm:mb-0 flex justify-between items-center"
    >
      <div
        className="text-2xl sm:text-3xl m-0"
      >
        <a
          className="flex items-center"
          href="/"
        >
          <img
            alt="logo"
            className="w-10 mr-1"
            src="test-file-stub"
          />
          Hashtag Coder
        </a>
      </div>
      <div
        className="sm:hidden"
      >
        <button
          className="hover:text-gray-500 focus:text-gray-500 focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-bars fa-w-14 "
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
    </div>
    <ul
      className="navigation bg-gray-200 rounded sm:bg-white p-2 sm:p-0 sm:flex animated fadeInRight hidden"
    >
      <li
        className="mb-2 sm:mb-0"
      >
        <a
          className="hoverable"
          href="/"
        >
          
        </a>
      </li>
      <div
        className="flex"
      >
        <li>
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-twitter fa-w-16 "
              data-icon="twitter"
              data-prefix="fab"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </li>
      </div>
    </ul>
  </header>
</div>
`;
